fm/bleh.user.js

```diff
@@ -19863,38 +19860,49 @@ let has_prompted_for_update = false;
 
             if (form)
                 panel.appendChild(form);
+
             if (notifications)
                 panel.appendChild(notifications);
+
             if (pagination)
                 panel.appendChild(pagination);
 
             page.structure.main.appendChild(panel);
 
-
             if (!notifications)
                 return;
 
             let notif_links = notifications.querySelectorAll('.inbox-notifications__item-link');
+
             notif_links.forEach((notification) => {
+                let link = notification.getAttribute('href');
+                if (link.endsWith('/obsessions/set') || link.endsWith('/listening-report/month')) return;
+
                 let avatar = notification.querySelector('.avatar');
                 let name = notification.querySelector('.inbox-notifications__item-description strong');
+                if (!name) return;
 
-                if (!name)
-                    return;
-
-                let name_text = return_name_from_avatar(avatar.querySelector('img'));
+                let avatar_img = avatar.querySelector('img');
+                if (!avatar_img) return;
 
+                let name_text = return_name_from_avatar(avatar_img);
+                let safeClassName = encodeURIComponent(name_text).replace(/%[0-9A-F]{2}/g, '_');
                 let badge = patch_avatar(avatar, name_text);
-                name.classList.add('notification-user-name', `user-status--bleh-${badge.type}`, `user-status--bleh-user-${name_text}`);
+
+                name.classList.add('notification-user-name',
+                    `user-status--bleh-${badge.type}`,
+                    `user-status--bleh-user-${safeClassName}`);
 
                 if (notification.classList.contains('inbox-notifications__item--highlight'))
-                    notification.classList.add('notification-user-name', `user-status--bleh-${badge.type}`, `user-status--bleh-user-${name_text}`);
+                    notification.classList.add('notification-user-name',
+                        `user-status--bleh-${badge.type}`,
+                        `user-status--bleh-user-${safeClassName}`);
```

Comment:
`safeClassName` is probably not needed anymore as we're already checking if this is a user before attempting to grab badges. only reason there'd be spaces is if last.fm somehow allowed that at one point (which they might have done, horror)

if so u should be able to change it to
```js
let name_text = sanitise(bla bla bla);
```
and change safeClassName references back to name_text (sanitise is a bleh function)